python/source/org/libreoffice/ide/eclipse/python/PythonProjectHandler.java

```diff
@@ -115,32 +119,48 @@ public void addOOoDependencies(IOOo pOoo, IProject pProject) {
-        //                pProject.open(null);
-        //                PluginLogger.debug("Project opened"); //$NON-NLS-1$
-        //            }
-        //
-        //            IProjectDescription description = pProject.getDescription();
-        //            String[] natureIds = description.getNatureIds();
-        //            String[] newNatureIds = new String[natureIds.length + 1];
-        //            System.arraycopy(natureIds, 0, newNatureIds, 0, natureIds.length);
-        //
-        //            // Adding the nature
-        //            newNatureIds[natureIds.length] = JavaCore.NATURE_ID;
-        //
-        //            description.setNatureIds(newNatureIds);
-        //            pProject.setDescription(description, null);
-        //            PluginLogger.debug(Messages.getString("Language.JavaNatureSet")); //$NON-NLS-1$
-        //
-        //        } catch (CoreException e) {
-        //            PluginLogger.error(Messages.getString("Language.NatureSettingFailed")); //$NON-NLS-1$
-        //        }
+        try {
+            if (!pProject.exists()) {
+                pProject.create(null);
+                PluginLogger.debug("Project created during language specific operation"); //$NON-NLS-1$
+            }
+
+            if (!pProject.isOpen()) {
+                pProject.open(null);
+                PluginLogger.debug("Project opened"); //$NON-NLS-1$
+            }
+
+            IProjectDescription description = pProject.getDescription();
+            String[] natureIds = description.getNatureIds();
+            String[] newNatureIds = new String[natureIds.length + 1];
+            System.arraycopy(natureIds, 0, newNatureIds, 0, natureIds.length);
+
+            // Adding the nature
+            newNatureIds[natureIds.length] = PYTHON_NATURE;
+
+            // Adding the buildCommand org.python.pydev.PyDevBuilder under buildSpec in .project file 
+            ICommand[] builders = description.getBuildSpec();
+
+            ICommand typesbuilderCommand = description.newCommand();
+            typesbuilderCommand.setBuilderName(PYTHON_BUILDER); //$NON-NLS-1$
+
+            //During the eclipse startup it keeps on adding the same buildCommand again
+            List<ICommand> list = Arrays.asList(builders);
+            if (!list.contains(typesbuilderCommand)) {
+                ICommand[] newCommands = new ICommand[builders.length + 1];
+                newCommands[0] = typesbuilderCommand;
+                System.arraycopy(builders, 0, newCommands, 1, builders.length);
+
+                description.setBuildSpec(newCommands);
+            }
+
+            description.setNatureIds(newNatureIds);
+            pProject.setDescription(description, null);
+            PluginLogger.debug(Messages.getString("Language.PythonNatureBuilderSet")); //$NON-NLS-1$
+
+        } catch (CoreException e) {
+            PluginLogger.error(Messages.getString("Language.NatureSettingFailed")); //$NON-NLS-1$
+        }
```

Comment:
This allows that in the '.project' file located under the Projct folder. The following set of line is added:-
&lt;buildCommand>
    &lt;name>org.python.pydev.PyDevBuilder&lt;/name>
    &lt;arguments>
    &lt;/arguments>
&lt;/buildCommand>

&lt;nature>org.python.pydev.pythonNature&lt;/nature>